persistence-api/src/test/java/io/stargate/db/cdc/CDCServiceTest.java

```diff
@@ -0,0 +1,109 @@
+package io.stargate.db.cdc;
+
+import static io.stargate.db.cdc.CDCServiceImpl.CDCConfig;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+
+import com.codahale.metrics.MetricRegistry;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.apache.cassandra.stargate.db.MutationEvent;
+import org.junit.jupiter.api.Test;
+
+public class CDCServiceTest {
+  @Test
+  public void shouldNotCallProducerWhenEventNotTrackedByCDC()
+      throws ExecutionException, InterruptedException {
+    ServiceBuilder builder = new ServiceBuilder().withTrackingByCDC(false);
+    CDCService service = builder.build();
+    service.publish(mock(MutationEvent.class)).get();
+    verify(builder.producer, times(0)).publish(any());
+  }
+
+  @Test
+  public void shouldNotCallProducerWhenUnhealthy() {
+    ServiceBuilder builder = new ServiceBuilder().withHealth(false);
+    CDCService service = builder.build();
+
+    ExecutionException runtimeEx =
+        assertThrows(
+            ExecutionException.class, () -> service.publish(mock(MutationEvent.class)).get());
+    assertThat(runtimeEx).hasRootCauseMessage("CDC producer marked as unhealthy");
+
+    verify(builder.producer, times(0)).publish(any());
+  }
+
+  @Test
+  public void shouldCallProducerAndReportSuccess() throws ExecutionException, InterruptedException {
+    ServiceBuilder builder = new ServiceBuilder();
+    CDCService service = builder.build();
+    service.publish(mock(MutationEvent.class)).get();
+    verify(builder.producer, times(1)).publish(any());
+    verify(builder.healthCheck, times(1)).reportSendSuccess();
+  }
+
+  @Test
+  public void shouldReportFailureWhenProducerFails() {
+    Exception ex = new Exception("Test error");
+    CompletableFuture<Void> future = new CompletableFuture<>();
+    future.completeExceptionally(ex);
+    ServiceBuilder builder = new ServiceBuilder().withProducerFuture(future);
+    CDCService service = builder.build();
+
+    ExecutionException runtimeEx =
+        assertThrows(
+            ExecutionException.class, () -> service.publish(mock(MutationEvent.class)).get());
+    assertThat(runtimeEx).hasCause(ex);
+  }
+
+  @Test
+  public void shouldTimeoutAndReportFailure() {
+    ServiceBuilder builder =
+        new ServiceBuilder().withProducerFuture(new CompletableFuture<>()).withTimeout(20);
+    CDCService service = builder.build();
+
+    ExecutionException runtimeEx =
```

Comment:
shouldn't this be wrapped in `await`? (https://github.com/awaitility/awaitility)

Reply:
I have no experience with `awaitability` library.

Using a new dependency for something that can be solved by `.get()` and unwrap exception seems unnecessary to me. We can create an utility method if we see the need.

Reply:
We are using this library for testing async code in java-driver and it has proven to be very robust. But in this situation you are right, `get()` is better